convert2rhel/actions/__init__.py

```diff
@@ -206,25 +211,94 @@ def run(self):
 
         self._has_run = True
 
-    def set_result(self, status=_NO_USER_VALUE, error_id=_NO_USER_VALUE, message=_NO_USER_VALUE):
+    @property
+    def result(self):
+        return self._result
+
+    @result.setter
+    def result(self, action_message):
+        """
+        Make sure the correct policy for result messages is used.
+        """
+        if not isinstance(action_message, ActionResult):
+            raise Exception()
+
+        self._result = action_message
+
+    def set_result(self, level, id=None, message=None):
+        """
+        Helper method that sets the resulting values for level, id and message.
+
+        :param id: The id to identify the result.
+        :type id: str
+        :param level: The status_code of the result .
+        :type: level: str
+        :param message: The message to be set.
+        :type message: str | None
+        """
+        if level not in ("ERROR", "OVERRIDABLE", "SKIP", "SUCCESS"):
+            raise KeyError("The level of result must be FAILURE, OVERRIDABLE, SKIP, or SUCCESS.")
+
+        result = ActionResult(level, id, message)
+        self.result = result
+
+    def add_message(self, level, id=None, message=None):
```

Comment:
Should we call this `set_message` instead? Just to apply the same naming as `set_result`

Reply:
It would be better to have the same naming scheme, I'll change this

Reply:
Note for people looking at this in the future, this function appends a message to the list of messages to be returned to the user while set_result() sets the one and only result from running the Action.  Therefore we kept this as add_message() to differentiate it for now.